### DISCUSSION TITLE
How Can I Globally Use  ISO C++ 20 for all new projects

### DISCUSSION THREAD

**Discussion by tmitchmachine** [eclipse-cdt/cdt#953] [General]
Hi I'm wondering if there is a way to globally apply dialect to my project so it works for all new projects. As a mac user I have issues with red squiggly lines under my code that aren't errors and I have to update this on new files is there a way to globally update it to for example

<img width="1440" alt="Screenshot 2024-12-04 at 8 29 56 PM (1)" src="https://github.com/user-attachments/assets/345c994c-055f-4205-8e6e-8115e576b3e3">

**Comment by jonahgraham** [comment]
Some additional discussion happened on this topic in https://github.com/eclipse-cdt/cdt/issues/956

**Comment by gvl2023** [comment]
You probably want to use other ways to manage your projects than the classic managed build system for several reasons. One is that the CDT C++ parser is not fully C++20 compliant and even with the dialect properly configured you will run into red squiggly lines if you use C++ 20 features it doesn't support. Look into using CMake projects with the LSP Editor/clangd backend.
